providers/FavouritesProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import {createContext, useEffect, useState} from "react";
+import {getFavourites, toggleFavourite} from "../async_storage/LocalStorage";
+
+interface FavouriteContextModel {
+    loading: boolean;
+    favourites: Array<number>;
+    toggle(id: number): void;
+    selected(id: number): boolean,
+}
+
+export const FavouritesContext = createContext<FavouriteContextModel>({
+    loading: true,
+    favourites: [],
+    toggle(id: number) {
+    },
+    selected(id: number) {
+        return true
+    },
+});
+
+export const FavouritesProvider = (props: any) => {
+
+    const [favourites, setFavourites] = useState<Array<number>>([]);
+
+    const [loading, setLoading] = useState<boolean>(false);
+
+    useEffect(() => {
+        setLoading(true);
+        getFavourites()
+            .then(setFavourites)
+            .catch(console.log)
+            .finally(() => setLoading(false))
+    }, []);
+
+    const toggle = (id: number): void => {
+        toggleFavourite(id).then(setFavourites).catch(console.log);
+    }
+
+    const selected = (id: number): boolean => favourites?.includes(id) ?? false;
```

Comment:
I think TS shouldn't shout if theis function would look like this: 

```suggestion
    const selected = (id: number): boolean => favourites.includes(id);
```